lib/internal/bootstrap/shadow_realm.js

```diff
@@ -0,0 +1,47 @@
+'use strict';
+
+// In ShadowRealm.
+
+const {
+  prepareShadowRealmExecution,
+} = require('internal/process/pre_execution');
+const {
+  BuiltinModule,
+} = require('internal/bootstrap/realm');
+const { setDefaultImportModuleDynamically } = require('internal/modules/esm/utils');
+
+BuiltinModule.setRealmAllowRequireByUsers([
+  'assert',
+  'assert/strict',
+  'buffer',
+  'console',
+  'constants',
+  'cypto',
+  'diagnostics_channel',
+  'events',
+  'module',
+  'path',
+  'path/posix',
+  'path/win32',
+  'readline',
+  'readline/promises',
+  'repl',
+  'stream',
+  'stream/consumers',
+  'stream/promises',
+  'stream/web',
+  'string_decoder',
+  'url',
+  'util',
+  'util/types',
+  'zlib',
+]);
+
+prepareShadowRealmExecution();
+
+// The handler for `ShadowRealm.prototype.importValue`.
+setDefaultImportModuleDynamically((specifier, assertions) => {
+  const { esmLoader } = require('internal/process/esm_loader');
```

Comment:
Can we emit a warning here about the leaks? (even though shadow realm is already behind flags, I guess an additional warning about something we already know wouldn't hurt...)